test/ViewExtensionLibraryTests/LibraryResourceProviderTests.cs

```diff
@@ -290,13 +293,212 @@ public void PackagedCustomNodeSearchElementLoadedType()
             var moq = new Mock<ICustomNodeSource>();
             var element = new CustomNodeSearchElement(moq.Object, info);
 
-            var provider = new NodeItemDataProvider(null);
+            var provider = new NodeItemDataProvider(new NodeSearchModel());
             var item = provider.CreateLoadedTypeItem(element);
             Assert.AreEqual(expectedQualifiedName, item.fullyQualifiedName);
             Assert.AreEqual(guid.ToString(), item.contextData);
             Assert.AreEqual(string.Empty, item.keywords);
             var url = new IconUrl(name, path, true);
             Assert.AreEqual(url.Url, item.iconUrl);
         }
+
+        [Test]
+        [Category("UnitTests")]
+        public void LibraryDataUpdatedEventRaised()
+        {
+            const string libraryDataUpdated = "libraryDataUpdated";
+            var timeout = 50; //50 milliseconds
+            var resetevent = new AutoResetEvent(false);
+
+            var model = new NodeSearchModel();
+            var controller = new Mock<IEventController>();
+            controller.Setup(c => c.RaiseEvent(It.IsAny<string>(), It.IsAny<object[]>())).Callback(() => resetevent.Set());
+
+            var disposable = LibraryViewController.SetupSearchModelEventsObserver(model, controller.Object, timeout);
+            controller.Verify(c => c.RaiseEvent(libraryDataUpdated, It.IsAny<object[]>()), Times.Never);
+
+            var d1 = MockNodeSearchElement("A", "B");
+            var d2 = MockNodeSearchElement("C", "D");
+            var d3 = MockNodeSearchElement("E", "F");
+            model.Add(d1.Object);
+            model.Add(d2.Object);
+            model.Add(d3.Object);
+            Assert.AreEqual(3, model.NumElements);
+
+            Assert.IsTrue(resetevent.WaitOne(timeout*3));
+            controller.Verify(c => c.RaiseEvent("libraryDataUpdated", "A, C, E"), Times.Once);
```

Comment:
both are same though, but yes I wanted to use the variable rather than a string.